nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -802,13 +838,7 @@ public static Map<String, Object> toMap(final Object value, final String fieldNa
         if (value instanceof Map) {
             final Map<?, ?> original = (Map<?, ?>) value;
 
-            boolean keysAreStrings = true;
-            for (final Object key : original.keySet()) {
-                if (!(key instanceof String)) {
-                    keysAreStrings = false;
-                    break;
-                }
-            }
+            boolean keysAreStrings = original.keySet().stream().allMatch(key -> key instanceof String);
```

Comment:
This should be reverted back. `DataTypeUtils` is extremely sensitive to performance as this method could potentially be called 100 times or more for a single Record, so we should not use `Stream` here - creation of a `Stream` has important performance impacts and should be reserved for non-performance-critical parts of the codebase.